src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -1111,6 +1107,8 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 				this._onProcessExit(error);
 			}
 		});
+		//TODO:find better place for this
+		this._onIconChanged.fire(this);
```

Comment:
The icon is changed in the constructor now, we may not need this anymore as it should be set immediately after the `TerminalInstance` is created?

Reply:
the background image isn't applied when I remove that.